docs/signature.md

```diff
@@ -0,0 +1,93 @@
+# Image Signature Specification
+
+**Version 0.1**
+
+## Introduction
+
+This document defines a detached container image signature object and signing methods.
+
+## Signature Format
+
+```js
+{
+    "critical": {/* required fields */
+        "identity": {/* identity reference */},
+        "image": {/* signed object reference */ },
+        "type": "..."
+    },
+    "optional": {/* optional metadata fields */}
+    }
+}
+```
+
+### Fields
+
+There are two top-level fields, **critical** (required) and **optional** (optional).
+
+#### `critical`
+
+**identity** (string):
+
+```js
+{
+    "docker-reference": imageName
+}
+```
+
+`imageName` per [V2 API](https://docs.docker.com/registry/spec/api/#/overview) Required.
```

Comment:
On Fri, Nov 18, 2016 at 02:26:30PM -0800, Miloslav Trmač wrote:
> > I'd name that product “product5/standalone:1.3”.
>
> Then we are back to tying the names to a particular namespace (design), and we have achieved nothing.

How does “describe the image in the image name, but don't inlclude the transport or signer” tie the names to a particular namespace?

> And, ultimately, making this interoperable with existing Docker references is a hard design constraint for us…

My understanding of Docker's naming policy is that [each “repository” (e.g. docker.io/library/ubuntu) has it's own root keys owned by the publisher][1].  So you'd run:

    $ tool pull docker.com/library/ubuntu:16:10

would map to:


    $ tool pull --namer docker.com/library/ubuntu --engine https://docker.com/library/ubuntu ubuntu:16:10

[1]: https://docs.docker.com/engine/security/trust/content_trust/#content-trust-operations-and-keys